src/container/cli.c

```diff
@@ -487,13 +487,131 @@ struct cont_open_args {
 	daos_handle_t		*hdlp;
 };
 
+struct pmap_refresh_cb_arg {
+	struct dc_pool		*pra_pool;
+	uint32_t		 pra_pm_ver;
+	uint32_t		 pra_retry_nr;
+};
+
+static int
+pmap_refresh_cb(tse_task_t *task, void *data)
+{
+	struct pmap_refresh_cb_arg	*cb_arg;
+	struct dc_pool			*pool;
+	uint32_t			 pm_ver;
+	uint64_t			 delay;
+	int				 rc = task->dt_result;
+
+	cb_arg = (struct pmap_refresh_cb_arg *)data;
+	pool = cb_arg->pra_pool;
+	if (rc)
+		goto out;
+
+	pm_ver = pool_map_get_version(pool->dp_map);
+	if (pm_ver < cb_arg->pra_pm_ver) {
+		if (cb_arg->pra_retry_nr > 10) {
+			/* basically this is impossible, or there is system
+			 * issue. Return EAGAIN just for code integrality.
+			 */
+			rc = -DER_AGAIN;
+			D_ERROR(DF_UUID": pmap_refresh cannot get required "
+				"version (%d:%d), try again later "DF_RC"\n",
+				DP_UUID(pool->dp_pool), pm_ver,
+				cb_arg->pra_pm_ver, DP_RC(rc));
+			goto out;
+		}
+		if (cb_arg->pra_retry_nr > 3)
+			delay = cb_arg->pra_retry_nr * 10;
```

Comment:
[Optional] The unit of delay is microsecond. So these retry, delay, and -DER_AGAIN seem a bit aggressive to me. Also, there's d_backoff_seq for the purpose here. But these are not showstoppers.

Reply:
Perhaps backoffs at the millisecond to second level would be better. Let's refine it later.